### PR TITLE
Firefox 42 compatibility (pair prog with @cgrebs)

### PR DESCRIPTION
**Pull request by magopian**
Fixes [bug 1214929](https://bugzilla.mozilla.org/show_bug.cgi?id=1214929)
Fixes [bug 1214926](https://bugzilla.mozilla.org/show_bug.cgi?id=1214926)
Fixes [bug 1214921](https://bugzilla.mozilla.org/show_bug.cgi?id=1214921)
Fixes [bug 1214912](https://bugzilla.mozilla.org/show_bug.cgi?id=1214912)

### CODE REVIEW

**Comment by magopian** [comment]
Ok so we have issues verifying those modifications: namely in the two following cases:
- https://bugzilla.mozilla.org/show_bug.cgi?id=1214929#c2
- https://bugzilla.mozilla.org/show_bug.cgi?id=1214912#c2

@andymckay, who would be able to give a review (and some ideas on the comments above)?

File: validator/testcases/javascript/predefinedentities.py [hunk 1]
```diff
@@ -9,7 +9,7 @@
 import instanceactions
 from jstypes import JSWrapper
 from validator.compat import FX40_DEFINITION
-from validator.constants import MDN_DOC
+from validator.constants import BUGZILLA_BUG, MDN_DOC
```
Comment (mstriemer) [inline validator/testcases/javascript/predefinedentities.py]: `BUGZILLA_BUG` isn't used.

**Comment by mstriemer** [comment]
The second issue should be solved by #335.

**Comment by magopian** [comment]
The second issue is indeed solved by #335, thanks @mstriemer you rock!

**Comment by magopian** [comment]
And we now also fixed the first issue, knowing that it may raise false positives (it'll flag any use of the `parseContentType` methods, whatever they relate to, even if they're not on the `nsINetUtil` service).

File: tests/compat/test_gecko42.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        """)
+        self.assert_compat_error()
+
+    def _test_nsIPermissionManager(self, method):
+        self.run_script_for_compat("""
+            var perms = Cc["@mozilla.org/permissionmanager;1"]
+                          .getService(Ci.nsIPermissionManager);
+            %s
+        """ % method)
+        self.assert_compat_error()
```
Comment (mstriemer) [inline tests/compat/test_gecko42.py]: This should probably look like:

```py
self.assert_silent()
self.assert_compat_error()
```

I know that's weird but the validation gets run twice. `self.assert_silent()`, `self.assert_failed()`, etc work against `self.err` and `self.assert_compat_error()` and `self.assert_compat_silent()` work against `self.compat_err`.

File: validator/testcases/javascript/entity_values.py [hunk 3]
```diff
@@ -171,6 +171,33 @@ def on_override(wrapper, arguments, traverser):
...
+@register_entity('nsIPermissionManager.remove')
+@register_entity('nsIPermissionManager.removeFromPrincipal')
+@register_entity('nsIPermissionManager.removeAll')
+@register_entity('nsIPermissionManager.testExactPermission')
+@register_entity('nsIPermissionManager.testExactPermissionFromPrincipal')
+@register_entity('nsIPermissionManager.testPermission')
+@register_entity('nsIPermissionManager.testPermissionFromPrincipal')
+def nsIPermissionManager(traverser):
+    traverser.err.warning(
+        err_id=('testcases_javascript_entity_values',
```
Comment (mstriemer) [inline validator/testcases/javascript/entity_values.py]: I think you need to set the version you want the compatibility error to happen for in this call.

**Pull request merged**